engine/src/main/java/org/terasology/rendering/nui/widgets/UIImageSlideshow.java

```diff
@@ -128,22 +116,73 @@ public void clean() {
         images = new ArrayList<>();
     }
 
-    protected boolean isActive() {
-        return active;
-    }
-
     /**
-     * Starts the slideshow.
+     * Starts automatic slideshow.
      */
     public void start() {
         active = true;
     }
 
     /**
-     * Stops the slideshow.
+     * Stops automatic slideshow.
      */
     public void stop() {
         active = false;
     }
 
+    /**
+     * Switches to next image of list.
+     */
+    public void nextImage() {
+        int size = images.size();
+        int prevIndex = index;
+        if (size > 0) {
+            if (index == size - 1) {
+                if (infinite) {
+                    index = 0;
+                } else {
+                    stop();
+                }
+            } else {
+                index++;
+            }
+            if (prevIndex != index) {
+                currentImage.set(images.get(index));
+            }
+        }
+    }
+
+    /**
+     * Switches to previous image of list.
+     */
+    public void prevImage() {
+        int size = images.size();
+        int prevIndex = index;
+        if (size > 0) {
+            if (index == 0) {
+                if (infinite) {
+                    index = size - 1;
+                } else {
+                    stop();
+                }
+            } else {
+                index--;
+            }
+            if (prevIndex != index) {
+                currentImage.set(images.get(index));
+            }
+        }
+    }
+
+    public float getSpeed() {
+        return speed;
+    }
+
+    public void setSpeed(float speed) {
+        this.speed = speed;
+    }
+
+    public List<UIImage> getImages() {
```

Comment:
### Option A
This would then return a list of `UIWidget`

### Option B
This would then return a list of `TextureRegion`